server/src/server.rs

```diff
@@ -72,6 +72,12 @@ impl Server {
 
                 let id = SessionId::try_from(session_id.clone())
                     .map_err(|_| Unauthorized::InvalidSessionId(session_id))?;
+
+                {
+                    let mut server = self.state.write().await;
+                    let _ = server.nodes.update_seen(id);
+                }
```

Comment:
Could be done on the beginning of the function for all `PacketKind`s.
`PacketKind::session_id(&packet);` function could be useful here